submissions/Roophee/friends-app/index.js

```diff
@@ -0,0 +1,221 @@
+                        <p>${item.location.country}</p>`
+    return cardItem;
+};
+
+const fetchData = (url) => fetch(url)
+.then((res) => {
+    if (res.status != 200) {
+        throw new Error('Network response was not OK');
+    }else{
+        return res.json();
+    };
+    })
+    .catch((error) => {
+        console.error('There has been a problem with your fetch operation:' , error)
+    })
+
+const getResult = (response) => {return response.results};
+
+const storeData = (result) => {
+    result.forEach((item) => {
+        RAW_DATA.push(item);
+        USER_STOR.set(item, createUserCards(item));
+    });
+};
+
+const getUserNode = (user) => {
+    return USER_STOR.get(user);
+};
+
+const createUserList = (usersData) => {
+    const documentFragment = document.createDocumentFragment();
+    usersData.forEach((item) => {documentFragment.appendChild(getUserNode(item))});
+    return documentFragment;
+};
+
+const appendUserList = (userList) => {
+    mainSection.appendChild(userList);
+};
+
+const updateUsers = (storeObj) => {
+    mainSection.innerHTML = '';
+    appendUserList(createUserList(storeObj))
+};
+
+const searchFilter = (user) => {
+    const fullName = `${user.name.first.toLowerCase()} ${user.name.last.toLowerCase()}`;
+    return fullName.includes(filterSettings.searchParam.trim());
+};
+
+const genderFilter = (user) => {
+    if(filterSettings.genderParam === 'all'){
+        return true;
+    } else{
+        return user.gender === filterSettings.genderParam;
+    };
+};
+
+const ageSort = (prev, next) => {
+    return prev.dob.age - next.dob.age;
+};
+
+const nameSort = (prev, next) => {
+    let nameA = prev.name.last.toLowerCase();
+    let nameB = next.name.last.toLowerCase();
+    if (nameA < nameB)
+        return -1;
+    if (nameA > nameB)
+        return 1;
+    return 0;
+};
+
+const applyFilters = (usersData, filter) => {
+    let filteredUsersData=[];
+    uncheckSortRadio();
+    if(filterSettings.filteredData.length < 1 && filterSettings.secondFilteredData.length < 1){
+        filteredUsersData = usersData.filter((user) => filter(user));
+        filterSettings.filteredData = filteredUsersData;
+        filterSettings.filterOne = filter.name;
+        return filteredUsersData;
+    }else if(filterSettings.filteredData.length >= 1 && filterSettings.secondFilteredData.length < 1 && filterSettings.filterOne === filter.name){
+        filteredUsersData = usersData.filter((user) => filter(user));
+        filterSettings.filteredData = filteredUsersData;
+        filterSettings.filterOne = filter.name;
+        return filteredUsersData;
+    }else if(filterSettings.filteredData.length >= 1 && filterSettings.secondFilteredData.length < 1 && filterSettings.filterOne !== filter.name){
+        filteredUsersData = filterSettings.filteredData.filter((user) => filter(user));
+        filterSettings.secondFilteredData = filteredUsersData;
+        filterSettings.filterTwo = filter.name;
```

Comment:
i don't see this props in initial `filterSettings`

`filterOne`
`filterTwo`

and what is the difference between these ones?